Depends build fails using mingw-w64 version 8.0.0-1 (GCC-10)

This is a heads up for future compatibility for if and when mingw-w64 gets updated. Shouldn't currently affect the official builds that are still on Bionic but needs to be kept in mind in case of future updates or for those that build for themselves.

When cross-compiling depends using mingw-w64 version 8.0.0-1 (GCC-10), build fails during the build of libzip with the following error:

`libtool: link: x86_64-w64-mingw32-gcc -pipe -O2 -I/home/g/GRC-Build/Gridcoin-Research/depends/x86_64-w64-mingw32 include -fPIC -I/home/g/GRC-Build/Gridcoin-Research/depends/x86_64-w64-mingw32/include -I/home/g/GRC-Build/Gridcoin-Research/depends/x86_64-w64-mingw32/include -o ziptool.exe source_hole.o ziptool.o  -L/home/g/GRC-Build/Gridcoin-Research/depends/x86_64-w64-mingw32/lib ../lib/.libs/libzip.a -lbz2 -lz`
`/usr/bin/x86_64-w64-mingw32-ld: ziptool.o:ziptool.c:(.text.startup+0x6f1): undefined reference to `__imp__snprintf.constprop.0'`

Downgrading back to mingw-w64 version 6.0.0-3 (GCC-8) from Debian Buster allows a successful build.

I also suspect version 7.0.0 (GCC-9) from Ubuntu Focul and Ubuntu Groovy are ok but have been unable to test them. I wouldn't be suprised to see these be updated at some point though as GCC-10 is already on their package managers and is already the default on Groovy.

Unfortunately both versions of mingw-w64 cant co-exist on the same system like GCC and G++ so the following packages have to be downgraded and held at the earlier versions to work around this issue (in my case to those available on Debain Buster, version numbers will vary on Ubuntu).

`g++-mingw-w64-i686=8.3.0-6+21.3~deb10u1`
`g++-mingw-w64-x86-64=8.3.0-6+21.3~deb10u1`
`g++-mingw-w64=8.3.0-6+21.3~deb10u1`
`gcc-mingw-w64-base=8.3.0-6+21.3~deb10u1`
`gcc-mingw-w64-i686=8.3.0-6+21.3~deb10u1`
`gcc-mingw-w64-x86-64=8.3.0-6+21.3~deb10u1`
`gcc-mingw-w64=8.3.0-6+21.3~deb10u1`
`mingw-w64-common=6.0.0-3`
`mingw-w64-i686-dev=6.0.0-3`
`mingw-w64-x86-64-dev=6.0.0-3`
`mingw-w64=6.0.0-3`

Build System:
WSL Debian Testing

`~$ lsb_release -da`
`Distributor ID: Debian`
`Description: Debian GNU/Linux bullseye/sid`
`Release: testing`
`Codename: bullseye`

Not-working
`~$ x86_64-w64-mingw32-gcc --version`
`x86_64-w64-mingw32-gcc (GCC) 10-posix 20200525`

Working
`~$ x86_64-w64-mingw32-gcc --version`
`x86_64-w64-mingw32-gcc (GCC) 8.3-posix 20190406`

--- a/depends/packages/libzip.mk
+++ b/depends/packages/libzip.mk
@@ -4,7 +4,7 @@ $(package)_download_path=https://libzip.org/download/
 $(package)_file_name=$(package)-$($(package)_version).tar.gz
 $(package)_sha256_hash=ab4c34eb6c3a08b678cd0f2450a6c57a13e9618b1ba34ee45d00eb5327316457
 $(package)_dependencies=zlib bzip2
-$(package)_patches=nonrandomopentest.c.patch
+$(package)_patches=nonrandomopentest.c.patch compat.h.patch
 
 
 define $(package)_set_vars
@@ -27,7 +27,8 @@ endif
 
 define $(package)_preprocess_cmds
   sed -i.old 's/\#  ifdef _WIN32/\#  if defined _WIN32 \&\& defined ZIP_DLL/' lib/zip.h && \
-  patch -p1 < $($(package)_patch_dir)/nonrandomopentest.c.patch
+  patch -p1 < $($(package)_patch_dir)/nonrandomopentest.c.patch && \
+  patch -p1 < $($(package)_patch_dir)/compat.h.patch
 endef
 
 define $(package)_config_cmds

--- a/depends/patches/libzip/compat.h.patch
+++ b/depends/patches/libzip/compat.h.patch
@@ -1,0 +1,15 @@
+--- old/lib/compat.h	2017-10-06 07:00:00.000000000 -0400
++++ new/lib/compat.h	2021-03-28 00:19:50.891360075 -0400
+@@ -92,9 +92,9 @@
+ #if defined(HAVE__OPEN)
+ #define open(a, b, c)	_open((a), (b))
+ #endif
+-#if defined(HAVE__SNPRINTF)
+-#define snprintf	_snprintf
+-#endif
++//#if defined(HAVE__SNPRINTF)
++//#define snprintf	_snprintf
++//#endif
+ #if defined(HAVE__STRDUP)
+ #if !defined(HAVE_STRDUP) || defined(_WIN32)
+ #undef strdup

